src/interface-validator.ts

```diff
@@ -58,6 +58,15 @@ export function getMismatchedFields(
       return (isOptional) ? [] : [valueKey];
     }
 
+    if (_.isArray(fieldSpecs[specName])) {
+      return values[valueKey].map((arrayValue, index) => {
+        return getMismatchedFields(
+          arrayValue,
+          fieldSpecs[specName][0] as Interface,
```

Comment:
Please add a array out-of-bound guard here. If `fieldSpecs[specName][0]` is `undefined`, throw a TypeError here. The message can be a JSON.stringified string of the whole `fieldSpecs` and the `specName` to help with debugging.